packages/SwingSet/docs/async.md

```diff
@@ -0,0 +1,91 @@
+# Asynchronous Considerations in SwingSet
+
+When working with SwingSet, it's crucial to understand how asynchronous operations interact with the system's execution model. A **vat** (a unit of isolated execution in SwingSet) may restart or terminate before an asynchronous operation completes. Although JavaScript's `await` keyword allows you to write asynchronous code that resembles synchronous code, you must be mindful of events that could prevent a promise from settling:
+
+- **Underlying condition never satisfied**: The condition required for the promise to resolve may never occur.
+- **Vat restarts or upgrades before settlement**: The promise may be severed if the vat restarts or upgrades before it settles.
```

Comment:
"the vat" is sort of ambiguous - there are typically 2 vats in question: the vat that exports a promise and the one that subscribes to it.

In some cases there's just 1 vat - promises that aren't exported. But those are sort of uninteresting here.

It's also possible to have more than 1 subscriber, so in general, there are N>=1 vats involved.